### ISSUE TITLE
Dependency constraint failure when trying to update to latest version of requests

### ISSUE THREAD

**Issue by federicobond** [DevCycleHQ/python-server-sdk#68]
Hi, when trying to update to the latest version of the `requests` library, I'm getting an error because the DevCycle SDK is a bit too restrictive in its dependencies declaration. Would it be possible to relax the constraint to requests>=2.31.0,<3.0.dev0? `requests` is pretty stable across minor versions.

```
  × No solution found when resolving dependencies:
  ╰─▶ Because devcycle-python-server-sdk==3.5.0 depends on
      requests>=2.31.0,<2.32.dev0 and you require requests==2.32.1,
      we can conclude that you require==0a0.dev0 and
      devcycle-python-server-sdk==3.5.0 are incompatible.
      And because you require devcycle-python-server-sdk==3.5.0, we can
      conclude that the requirements are unsatisfiable.
```

_Edit: I'm using `uv pip` to install dependencies here. Results with other tools may vary_

**Comment by kaushalkapasi** [comment]
Hey @federicobond. let me double check with the team but I think what you've suggested is a reasonable update to the version!

**Comment by federicobond** [comment]
Thanks @kaushalkapasi  🙌

**Comment by kaushalkapasi** [comment]
Hey @federicobond, want to checkout the PR https://github.com/DevCycleHQ/python-server-sdk/pull/69 and let me know if this change will work? We'd made accidentally restricted the request library to be narrowly versioned in our original configuration which only allowed `2.31.*` instead of allowing `>= 2.31`.

**Comment by kaushalkapasi** [comment]
Hey @federicobond we've released `v3.5.1` of the SDK with some of the dependency changes. Let us know if this works for you! https://github.com/DevCycleHQ/python-server-sdk/releases/tag/v3.5.1 

Thanks.

**Comment by federicobond** [comment]
Yes, this works perfectly. Thank you for the quick response.
